tt_eager/tt_lib/CMakeLists.txt

```diff
@@ -0,0 +1,27 @@
+
+set(TT_LIB_SRCS
+    ${CMAKE_CURRENT_SOURCE_DIR}/csrc/tt_lib_bindings.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/csrc/type_caster.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/csrc/tt_lib_bindings_tensor.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/csrc/tt_lib_bindings_tensor_composite_ops.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/csrc/tt_lib_bindings_tensor_backward_ops.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/csrc/tt_lib_bindings_tensor_pytensor.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/csrc/tt_lib_bindings_tensor_dm_ops.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/csrc/tt_lib_bindings_tensor_custom_bmm_ops.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/csrc/tt_lib_bindings_tensor_xary_ops.cpp
+)
+
+add_subdirectory(${CMAKE_SOURCE_DIR}/tt_metal/third_party/pybind11 ${CMAKE_BINARY_DIR}/pybind11)
+
+pybind11_add_module(tt_lib SHARED ${TT_LIB_SRCS})
```

Comment:
I think I put some things to potentially research from scikit / pybind11 authors. Check those out and let me know what you think

Reply:
yep, saw those, will reference to try to get pybind11 up and running